src/vnsw/agent/test-xml/test_xml.cc

```diff
@@ -240,36 +243,31 @@ bool AgentUtXmlTest::ReadXml() {
 }
 
 bool AgentUtXmlTest::Load() {
-    struct stat s;
-    if (stat(file_name_.c_str(), &s)) {
-        cout << "Error <" << strerror(errno) << "> opening file "
-            << file_name_ << endl;
+    boost::system::error_code ec;
+    boost::filesystem::path file_path(file_name_);
+    uintmax_t file_size = boost::filesystem::file_size(file_path, ec);
+    if (ec) {
+        cout << "Error <" << ec << "> opening file" << file_name_ << endl;
         return false;
     }
 
-    int fd = open(file_name_.c_str(), O_RDONLY);
-    if (fd < 0) {
-        cout << "Error <" << strerror(errno) << "> opening file "
-            << file_name_ << endl;
-        return false;
-    }
-
-    char data[s.st_size + 1];
-    if (read(fd, data, s.st_size) < s.st_size) {
-        cout << "Error <" << strerror(errno) << "> reading file "
-            << file_name_ << endl;
-        close(fd);
+    std::vector<char> data(file_size + 1, 0);
+    try {
```

Comment:
I think that we are not allowed to use exceptions in Agent code. Additionally, I think that `fstream` by default does not throw any exception.